lib/src/main/java/com/icapps/architecture/utils/caching/Cached.kt

```diff
@@ -0,0 +1,39 @@
+package com.icapps.architecture.utils.caching
+
+import kotlin.reflect.KProperty
+
+/**
+ * @author Maarten Van Giel
+ * A utility class to create variables that expire (return null) after a while.
+ * Usage: `var cachedValue: T? by Cached(CACHE_VALIDITY_MS)`
+ */
+class Cached<T>(private val cacheValidity: Long = DEFAULT_CACHE_DURATION) {
+
+    companion object {
+        const val DEFAULT_CACHE_DURATION = 60000L
+    }
+
+    private var cache: T? = null
+    private var timestamp: Long? = null
+
+    operator fun getValue(thisRef: Any, property: KProperty<*>): T? {
+        synchronized(this) {
+            return timestamp?.let {
+                if (System.currentTimeMillis() - it <= cacheValidity) {
```

Comment:
currentTimeMillis is a bad clock to use for relative durations. Time update can shift this value. We should use a better clock. On android we have the SystemClock.elapsedRealtime() which provides more correct values (based on the kdoc of this class)